src/org/loklak/data/ElasticsearchClient.java

```diff
@@ -988,4 +990,64 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
                 AggregationBuilders.sum("sum_probability").field(probabilityField)
             );
     }
+
+    public HashMap<String, HashMap<String, Double>> classifierScore(String index, String classifierName, ArrayList<String> classes, String startDate, String endDate) {
+        if (startDate == null && endDate == null) {
+            return classifierScore(index, classifierName, classes);
+        }
+
+        SearchRequestBuilder request = elasticsearchClient.prepareSearch(index)
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setQuery(QueryBuilders.matchAllQuery())
+            .setFrom(0)
+            .setSize(0)
+            .addAggregation(getClassifierAggregation(classifierName, startDate, endDate));
+
+        SearchResponse response = request.execute().actionGet();
+        Range aggrs = response.getAggregations().get("by_date");
+
+        HashMap<String, HashMap<String, Double>> aggregatedMap = new HashMap<>();
+
+        for (Range.Bucket bkt : aggrs.getBuckets()) {
+            Terms termAggrs = bkt.getAggregations().get("by_class");
```

Comment:
ok, this answer says this is best scope :)
https://stackoverflow.com/questions/407255/difference-between-declaring-variables-before-or-in-loop